syft/frameworks/torch/tensors/interpreters/large_precision.py

```diff
@@ -0,0 +1,226 @@
+        return torch.from_numpy(result.reshape(self.child.shape[:-1]).astype(np.float32))
+
+    @staticmethod
+    def create_tensor_from_numpy(ndarray, **kwargs):
+        """Decompose a NumPy array into an array of numbers that represent such tensor with the required precision.
+
+        Typically this private method is called on the result of an operation.
+        """
+        internal_type = kwargs["internal_type"]
+        internal_precision = type_precision[internal_type]
+        original_shape = ndarray.shape
+        result = []
+        # refs_ok is necessary to enable iterations of reference types.
+        for x in np.nditer(ndarray, flags=["refs_ok"]):
+            result.append(
+                LargePrecisionTensor._split_number(x.item(), internal_precision, internal_type)
+            )
+        new_shape = original_shape + (len(max(result, key=len)),)
+        result = np.array(result).reshape(new_shape)
+        return torch.tensor(result, dtype=internal_type)
+
+    @staticmethod
+    def _split_number(number, bits, internal_type) -> list:
+        """Splits a number in numbers of a smaller power.
+
+        Args:
+            number (int): the number to split.
+            bits (int): the bits to use in the split.
+
+        Returns:
+            list: a list of numbers representing the original one.
+
+        """
+        if not number:
+            return [number]
+
+        if number > 0:
+            sign = 1
+        else:
+            assert (
+                internal_type != torch.uint8
+            ), "Negative LargePrecisionTensors cannot be represented with uint8"
+            sign = -1
+            number = number * (-1)
+
+        base = 2 ** bits
+        number_parts = []
+        while number:
+            number, part = divmod(number, base)
+            number_parts.append(part * sign)
+        return number_parts[::-1]
+
+    @staticmethod
+    def _internal_representation_to_large_ints(tensor, precision) -> np.array:
+        """Creates an numpy array containing the objective large numbers."""
+        ndarray = tensor.numpy()
+        ndarray = ndarray.reshape(-1, ndarray.shape[-1])
+        result = []
+        for elem in ndarray:
+            result.append(LargePrecisionTensor._restore_large_number(elem, precision))
+        return np.array(result).reshape(tensor.shape[:-1])
+
+    @staticmethod
+    def _restore_large_number(number_parts, bits) -> int:
+        """Rebuilds a number from a numpy array.
+
+        Args:
+            number_parts (ndarray): the numpy array of numbers representing the original one.
+            bits (int): the bits used in the split.
+
+        Returns:
+            Number: the large number represented by this tensor
+        """
+        base = 2 ** bits
+        n = 0
+        for x in number_parts.tolist():
+            n = n * base + x
+        return n
+
+
+type_precision = {
```

Comment:
Maybe add a comment again to explain why you needed to do -1

Reply:
Changed the type sizes to be what the platform provides. The -1 is applied when computing the internal size (comments added to the source to explain it)